member/src/main/scala/com/improving/app/member/Main.scala

```diff
@@ -1,29 +1,59 @@
 package com.improving.app.member
 
-import com.improving.app.member.domain.Member
-import kalix.scalasdk.Kalix
-import org.slf4j.LoggerFactory
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.scaladsl.Behaviors
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import com.improving.app.member.api.{MemberServiceHandler, MemberServiceImpl}
+import com.typesafe.config.ConfigFactory
+import com.typesafe.scalalogging.StrictLogging
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+import scala.io.StdIn
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
 
 // This class was initially generated based on the .proto definition by Kalix tooling.
 //
 // As long as this file exists it will not be overwritten: you can maintain it yourself,
 // or delete it so it is regenerated as needed.
 
-object Main {
+object Main extends App with StrictLogging {
 
-  private val log = LoggerFactory.getLogger("com.improving.app.member.Main")
+  private val conf = ConfigFactory
+    .parseString("akka.http.server.preview.enable-http2 = on")
+    .withFallback(ConfigFactory.defaultApplication())
 
-  def createKalix(): Kalix = {
-    // The KalixFactory automatically registers any generated Actions, Views or Entities,
-    // and is kept up-to-date with any changes in your protobuf definitions.
-    // If you prefer, you may remove this and manually register these components in a
-    // `Kalix()` instance.
-    KalixFactory.withComponents(
-      new Member(_))
-  }
+  implicit val system: ActorSystem[_] = ActorSystem[Nothing](Behaviors.empty, "MemberSystem", conf)
+  implicit val ec: ExecutionContext = system.executionContext
+
+  private def run(): Future[Http.ServerBinding] = {
+
+    val service: HttpRequest => Future[HttpResponse] = MemberServiceHandler(new MemberServiceImpl())
+
+    val bound: Future[Http.ServerBinding] = Http(system)
+      .newServerAt(interface = "127.0.0.1", port = 8080)
+      //.enableHttps(serverHttpContext)
+      .bind(service)
+      .map(_.addToCoordinatedShutdown(hardTerminationDeadline = 10.seconds))
 
-  def main(args: Array[String]): Unit = {
-    log.info("starting the Kalix service")
-    createKalix().start()
+    println(s"Server online at http://localhost:8080/\nPress RETURN to stop...")
```

Comment:
You can follow my main as *I think* mine is the more correct way that doesn't require the read line

Reply:
fixed